### PR TITLE
Prevent adding yourself as a peer

### PR DESCRIPTION
**Pull request by StriderDM**
## Description
Updated add_peers_to_comms to prevent adding yourself as a peer

## Motivation and Context
To be able to use a single configuration file for all base nodes

## How Has This Been Tested?
cargo test --all --all-features

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/builder.rs [hunk 1]
```diff
@@ -905,14 +905,23 @@ async fn setup_wallet_comms(
 }
 
 async fn add_peers_to_comms(comms: &CommsNode, peers: Vec<Peer>) -> Result<(), String> {
+    let node_identity = comms.node_identity().clone();
```
Comment (sdbondi) [inline applications/tari_base_node/src/builder.rs:908]: No need to clone here - a reference should do

File: applications/tari_base_node/src/builder.rs [hunk 2]
```diff
@@ -905,14 +905,23 @@ async fn setup_wallet_comms(
...
     for p in peers {
         let peer_desc = p.to_string();
         info!(target: LOG_TARGET, "Adding seed peer [{}]", peer_desc);
-        comms
-            .peer_manager()
-            .add_peer(p)
-            .await
-            .map_err(|e| format!("Could not add peer {} to comms layer: {}", peer_desc, e))?;
+
+        if p.public_key != node_identity.public_key().to_owned() {
```
Comment (sdbondi) [inline applications/tari_base_node/src/builder.rs:913]: As clippy says :) - 
```suggestion
        if &p.public_key != node_identity.public_key() {
```

Comment (sdbondi) [reply applications/tari_base_node/src/builder.rs:913]: Rather check for this case and use `continue` 
```suggestion
        if node_identity.public_key() == &p.public_key {
        info!(
                target: LOG_TARGET,
                "Attempting to add yourself [{}] as a seed peer to comms layer, ignoring request", peer_desc
            );
            continue;
            }
```

File: applications/tari_base_node/src/builder.rs (hunk 1 above)
Comment (StriderDM) [reply applications/tari_base_node/src/builder.rs:908]: Think the value will be freed after the call, will add it directly to the comparison

Comment (sdbondi) [reply applications/tari_base_node/src/builder.rs]: Nope :)

**Pull request merged**